src/Core/Package.vala

```diff
@@ -709,10 +708,22 @@ public class AppCenterCore.Package : Object {
                     bool is_bigger = (icon_width > current_size && current_size < pixel_size);
                     bool has_better_dpi = (icon_width == current_size && current_scale < icon_scale && scale_factor <= icon_scale);
                     if (is_bigger || has_better_dpi) {
-                        var file = File.new_for_uri (_icon.get_url ());
-                        icon = new FileIcon (file);
-                        current_size = icon_width;
-                        current_scale = icon_scale;
+                        var server_file = File.new_for_uri (_icon.get_url ());
+                        var local_file = File.new_for_path (
+                            Path.build_filename (Environment.get_tmp_dir (), server_file.get_basename ())
+                        );
+
+                        // FIXME: Copy this async
+                        try {
+                            server_file.copy (local_file, FileCopyFlags.OVERWRITE, null, null);
+
+                            icon = new FileIcon (local_file);
+                            current_size = icon_width;
+                            current_scale = icon_scale;
+                        } catch (Error e) {
+                            critical ("Unable to download remote icon: %s", e.message);
+                        }
```

Comment:
I get random freezes of AppCenter with a warning in the console:

```sh
** (io.elementary.appcenter:9522): CRITICAL **: 23:14:38.142: Package.vala:724: Unable to download remote icon: Not Found
```

The issue could be related to this code.